sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/RegistrarClient.java

```diff
@@ -0,0 +1,212 @@
+import com.azure.android.core.util.CancellationToken;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static com.azure.android.communication.chat.BuildConfig.PLATFORM;
+import static com.azure.android.communication.chat.BuildConfig.PLATFORM_UI_VERSION;
+import static com.azure.android.communication.chat.BuildConfig.PUSHNOTIFICATION_REGISTRAR_SERVICE_URL;
+import static com.azure.android.communication.chat.BuildConfig.PUSHNOTIFICATION_REGISTRAR_SERVICE_TTL;
+import static com.azure.android.communication.chat.BuildConfig.PUSHNOTIFICATION_APPLICATION_ID;
+import static com.azure.android.communication.chat.BuildConfig.PUSHNOTIFICATION_TEMPLATE_KEY;
+
+/**
+ * The registrar client interface
+ */
+public class RegistrarClient {
+    private static final String SKYPE_TOKEN_HEADER = "X-Skypetoken";
+    private static final String USER_AGENT_HEADER = "User-Agent";
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+
+    private static final String CONTENT_TYPE_JSON = "application/json";
+    private static final String NODE_ID = "";
+
+    private final ClientLogger logger = new ClientLogger(RegistrarClient.class);
+    private final HttpClient httpClient;
+    private final JacksonSerder jacksonSerder;
+
+    private String registrationId;
+    private boolean requestResult;
+
+    private static class ClientDescription {
+        @JsonProperty(value = "languageId")
+        private String languageId;
+
+        @JsonProperty(value = "platform")
+        private String platform;
+
+        @JsonProperty(value = "platformUIVersion")
+        private String platformUIVersion;
+
+        @JsonProperty(value = "appId")
+        private String applicationId;
+
+        @JsonProperty(value = "templateKey")
+        private String templateKey;
+    }
+
+    private static class FcmTransport {
+        @JsonProperty(value = "context")
+        private String context;
+
+        @JsonProperty(value = "creationTime")
+        private String creationTime;
+
+        @JsonProperty(value = "path")
+        private String path;
+
+        @JsonProperty(value = "ttl")
+        private String ttl;
+    }
+
+    private static class Transports {
+        @JsonProperty(value = "FCM")
+        private List<FcmTransport> fcm;
+    }
+
+    private static class RegistrarRequestBody {
+        @JsonProperty(value = "clientDescription")
+        private ClientDescription clientDescription;
+
+        @JsonProperty(value = "nodeId")
+        private String nodeId;
+
+        @JsonProperty(value = "registrationId")
+        private String registrationId;
+
+        @JsonProperty(value = "transports")
+        private Transports transports;
+    }
+
+    RegistrarClient() {
+        this.httpClient = new OkHttpAsyncHttpClientBuilder().build();
```

Comment:
We probably should use [this](https://github.com/Azure/azure-sdk-for-android/blob/71214038c6d7ba05b3d2a4b11ceec65771107ef5/sdk/core/azure-core-http/src/main/java/com/azure/android/core/http/HttpClient.java#L42) to keep the HTTP client library pluggable:
```suggestion
        this.httpClient = HttpClient.createDefault();
```